### PR TITLE
Fix placement of dots in the list

### PR DESCRIPTION
**Pull request by venables**
On a result set with 9 pages, and the default "adjacent" value, the
pagination would look like "1 2 3 4 5 6 7 ... 8 9"

This ensures we don't ever have dots in between consecutive numbers

### CODE REVIEW

**Comment by brantwills** [comment]
Hey venables this is Jamtastic!
I'll begin reviewing the code and get this is as soon as possible

**Comment by brantwills** [comment]
Alrighty I had a change to review,
I completely agree this is a bug and now that I see it man is it annoying!
The notes below are more for my own benefit but please feel free to comment

-----------

Ventables proposed changes work wonderfully using the following logic:
1) We first create the list without any dots at all
2) We then create a temp list in memory
3) We then roll over the original list and check where a value shifts 
4) We add each item to the temp list - adding dots where a shift exists
5) We replace the original list with the temp list

I began examining if it was possible to avoid the whole temp list idea
It felt like there had to be a way to achieve this without creating the list all over again just for the dots
I discovered if we just pass in the "previous" and "next" range values we can directly affect the dots logic!

In short we just add some logic checks in the "addFirst" and "addLast" methods
If the next range value is '3' in addFirst we do not need dots (since 1 and 2 are hardcoded)
If the prev range value is equal to 'pageCount - 2' in addLast we do not need dots  

I have generated the code and will push it to master shortly - closing this pull request
I've tested this as best I can and did not see any immediate issues 
I'm worried there might be some funk with "addLast" but testing was ok
If any issues are found we can log them as a new issue or pull request item

Side note.. makes me think its time to add some unit testing .. ugh maturity!
